src/librustc_trans/mir/block.rs

```diff
@@ -281,7 +281,7 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
 
             mir::TerminatorKind::Assert { ref cond, expected, ref msg, target, cleanup } => {
                 let cond = self.trans_operand(&bcx, cond).immediate();
-                let mut const_cond = common::const_to_opt_uint(cond).map(|c| c == 1);
+                let mut const_cond = common::const_to_opt_u128(cond, false).map(|c| c == 1);
```

Comment:
Heh, this only wants one bit (`cond` has type `i1`). Not sure if there's a nicer way to do this though.